src-built-in/components/toolbarRefactored/toolbarHotkeys.ts

```diff
@@ -0,0 +1,109 @@
+const keys = FSBL.Clients.HotkeyClient.keyMap;
+
+/**
+ * Helper so I didn't have to rewrite the same 3 lines of code 6
+ * times.
+ * @param err Unlikely error; would be coming in when our hotkey is registered.
+ */
+const handleHotkeyRegistrationError = (err:any) => {
+	if (err) {
+		FSBL.Clients.Logger.error(`HotkeyClient.addGlobalHotkey failed, error:`, err);
+	}
+}
+/**
+	 * Function to bring toolbar to front (since dockable toolbar can be hidden)
+	 * The search input box will be open and any previous results will be displayed
+	 * @param {boolean} focus If true, will also focus the toolbar
+	 * @TODO Evaluate whether focus is necessary or if it's just vestigial.
+	 */
+const bringToolbarToFront = (focus = false) => {
+	finsembleWindow.bringToFront(null, (err:any) => {
+		if (err) {
+			FSBL.Clients.Logger.error(`bringToolbarToFront failed, error:`, err);
+		}
+
+		if (focus) {
+			finsembleWindow.focus();
+			// @todo when the search component is finished, dispatch the action
+			// that focuses the search field
+		}
+	});
+}
+
+ 	/**
+* Hides the toolbar
+*/
+const hideToolbar = () => {
+ finsembleWindow.blur();
+ finsembleWindow.hide();
```

Comment:
Shouldn't this be in the callback to `blur`? I know they're on a queue internally, but we shouldn't presume that knowledge here.

Reply:
No sir; we want it to hide immediately. We blur just so that any open menus will also.

Reply:
I'm a dumb-dumb.